### ISSUE TITLE
Incorrect behaviour in DbManager

### ISSUE THREAD

**Issue by StalkAlex** [yiisoft/yii2#13501]
### What steps will reproduce the problem?
Trying to create permission
```
$permission = $auth->createPermission('test');
$permission->ruleName = 'TestRule';
$auth->add($permission)
```
data of rule is ```O:24:"app\rules\TestRule":3:{s:4:"name";s:14:"TestOnly";s:9:"createdAt";i:1485582849;s:9:"updatedAt";i:1485582849;}```

### What do you get instead?
Error
```bash
PHP Warning 'yii\base\ErrorException' with message 'unserialize() expects parameter 1 to be string, resource given'

in /vagrant/vendor/yiisoft/yii2/rbac/DbManager.php:634
```
Here's screen. As it seen data variable isn't sting anymore but resource stream.
![test](https://cloud.githubusercontent.com/assets/2943154/22559100/e7085e4e-e991-11e6-9176-02b529ff6636.png)

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.11
| PHP version      | 7.1.1
| Operating system | CentOS 7
| Database | PostgreSQL 9.6

**Comment by StalkAlex** [comment]
There is no such bug in 2.0.10, so it's not db related problem.

**Comment by samdark** [comment]
Probably. Can you find out which commit breaks it using `git bisect`?

**Comment by StalkAlex** [comment]
Yep. I check.

**Comment by StalkAlex** [comment]
@samdark  found commit that causes this bug https://github.com/yiisoft/yii2/commit/23790272dc561aacf2070cbbda396f49e44cbb7d

**Comment by cebe** [comment]
To fix it a check should be added that will read all data from the stream. possibly via http://php.net/manual/en/function.stream-get-contents.php

**Comment by StalkAlex** [comment]
Like this? https://github.com/StalkAlex/yii2/commit/5a76ef110e99d377b1f79a2b814656da45a7b610 Checked in own project, it worked.

**Comment by samdark** [comment]
Looks OK.

**Comment by cebe** [comment]
There is now a 2.0.11.2 release including this fix. https://github.com/yiisoft/yii2/blob/2.0.11.2/framework/CHANGELOG.md#20112-february-08-2017

**Comment by StalkAlex** [comment]
Good news, thanks!

**Comment by simonevallana** [comment]
This problem remain also in the 2.0.11.2 version in the method loadFromCache() in DbManager class (line 990). Could you fix it? Thank you

**Comment by samdark** [comment]
@simonevallana it's fixed in 2.0.12 which isn't relased yet.

**Comment by simonevallana** [comment]
Perfect, thank you.

**Comment by cebe** [comment]
turns out RBAC tests do not cover pgsql and sqlite, even though they suggest they should.

**Comment by cebe** [comment]
fixed by fa4d762..67b9a57
